embedded/store/immustore_test.go

```diff
@@ -41,6 +41,17 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func immustoreClose(t *testing.T, immuStore *ImmuStore) {
+	err := immuStore.Close()
+	if !t.Failed() {
+		require.NoError(t, err)
```

Comment:
I'm wondering if error assertion does it work even when the method is deferred..

Reply:
Yes, I used that before to catch errors with session leaks.

And for that reason I also only check if the test did not fail before - in many cases when some requirement fails and breaks the test in the middle we did not yet cleanup properly and the error here was heavily polluting the test error report.